CMakeLists.txt

```diff
@@ -389,19 +383,33 @@ set(CMAKE_CUDA_STANDARD ${QUDA_CXX_STANDARD})
 set(CMAKE_CUDA_STANDARD_REQUIRED True)
 mark_as_advanced(CMAKE_CUDA_HOST_COMPILER)
 
-set(CMAKE_CUDA_DISABLE_XCOMPILER_WARNINGS
-    "-Wno-unknown-pragmas,-Wno-unused-function,-Wno-unused-local-typedef,-Wno-unused-private-field")
+include(CheckLanguage)
+check_language(CUDA)
+
+if(${CMAKE_CUDA_COMPILER} MATCHES "nvcc")
+  set(QUDA_CUDA_BUILD_TYPE "NVCC")
+  message(STATUS "CUDA Build Type: ${QUDA_CUDA_BUILD_TYPE}")
+endif()
+
+if(${CMAKE_CUDA_COMPILER} MATCHES "clang")
```

Comment:
Add a comment: I noticed that what I assume are `nvcc`-specific flags, like `-lineinfo`, were moved down to the `CMakeLists.txt` in the `lib` subdirectory. This seems like a potential avenue for issues in the future. Is there any reason the logic to add `-lineinfo`, etc, for `nvcc` builds was moved there and can't be brought back to this file?

Reply:
So right now I think those are guarded with the generator expression. The logic may be that this is a property of the library only? Ultimately this may need further factoring as we cope with more compilers...

Reply:
Good point: Long term I want to move target specific stuff in the target directories and then make sure to propagate them accordingly. Targets that need nvcc specific options should only live in the `lib/` and `target/cuda` directories so I am not too worried.